### ISSUE TITLE
Unrealized stuff and bugs

### ISSUE THREAD

**Issue by EastRane** [boiscljo/SBA#34]
Using this plugin for about 2 months now, thanks for it! Some problems have appeared during this time:
- The plugin gives the option to auto swap swords when buying a better quality sword, but doesn't do it with pickaxe and axe. It should, as it seems to me.
- The ability to disable downgrading of pickaxes/axes is not implemented. As well as the ability to configure all types of pickaxes/axes that will be given to a player when downgrading pickaxes/axes (personally, I do not need a gold pickaxe, which is hard-coded in the plugin code). As far as I understand, the armor material for downgrading can't be customized either
- It would also be nice to block the ability to move sword/tools in slots (to be customizable in the config file), not sure if it's already implemented. And it should be customizable in which slot to put this items after respawn etc (now it's hard-codedd like pickaxe is slot 7, axe is slot 8...)
- When I translate the internal NPC strings (games-inventory) and then do /sba generate, my string translation resets.
- I'm not quite sure I'm getting this right, but it seems like the values for fireballs and TNT are synchronized? In terms of knockback and such. Seems to me they should be set up separately. And it's pretty hard to configure fireballs for me, maybe it would be much better if you comment some parameters in config etc, because it doesn't look like hypixel's
- The implementation of spectating mode through adventure mode is very wrong imo. Since the plugin gives an invisibility effect, players can see the particles. In addition, the players see the items that are in the hands of the spectators. Plus, spectators in adventure mode can obscure arrows and other things, or what? At the same time, I don't like spectators in direct vanilla spectator mode, so it's kind of sad. Also, in spectator mode I can teleport to players who are at another world (out of current arena) xd.
- Also I'm not sure if this is to you or the developer of the bedwars plugin itself, but the villagers don't exist in spectator mode (to spectators of arena).
- I'm not sure why and I'm not sure if I can give some additional info, but for some reason all floating spawners in arenas just break after first match on server. They show "Spawner is full", but spawners still generate items, idk why. If I enable spawner holograms in SBW plugin, it shows spawner generation timer just well (in SBW holo, not SBA), but then there's an error when spectator enters arena (i guess this is why I shouldn't enable both holo settings at two plugin at the same time xd). Maybe I did something wrong in the config idk
- When ugrading internal plugin's files (after plugin's update), it deletes my changes in languages files.

All of this was in SBW 0.2.29 and SBA 1.5.9 and all of this remained in 0.2.30 & 1.5.11.

Paper 1.20.1.
`Bukkit Plugins: Authy, BedWars, Citizens, DecentHolograms, Essentials, FastAsyncWorldEdit, LuckPerms, MdMotd, Multiverse-Core, NoOffhand, OldCombatMechanics, PlaceholderAPI, ProtocolLib, RegenManager, SBA, SkinsRestorer, spark, SuperLobby, TabList, Vault, WhitelistByTime, YamipaPlugin`
SBA config: https://pastebin.com/hye06mJz
SBW config: https://pastebin.com/TMk6ffuj

**Comment by boiscljo** [comment]
When ugrading internal plugin's files (after plugin's update), it deletes my changes in languages files.

You are using /sba upgrade ,which reset everything, most of the time doing /sba cancel is better and old config will work just fine.

**Comment by boiscljo** [comment]
> I'm not quite sure I'm getting this right, but it seems like the values for fireballs and TNT are synchronized? In terms of knockback and such. Seems to me they should be set up separately. And it's pretty hard to configure fireballs for me, maybe it would be much better if you comment some parameters in config etc, because it doesn't look like hypixel's


The way the configs are generated I cannot put comments in there, but it will be added on the FAQ. It won't be like that server. I'm not even sure how the parameters affect the jump. But in short:
- The closer you are to the explosion the further you get thrown (explosion lose strength with distance)
- detection-distance is the range in which entities will be affected
- acceleration-y adds speed vertically (think of it like the inverse of gravity) a= a+acceleration-y
- reduce-y (Divive acceleration by it) a = a - reduce-y
- launch-multiplier is a multiplier in all directions so a = a*launch-multiplier

I'm not sure why they are together, it was like the when @pronze owned the plugin, I just took over.

**Comment by boiscljo** [comment]
>When I translate the internal NPC strings (games-inventory) and then do /sba generate, my string translation resets.

SBA generate completely reset games-inventory with the new games, I guess that also affect the npc somehow, where were those strings located

**Comment by boiscljo** [comment]
>Also I'm not sure if this is to you or the developer of the bedwars plugin itself, but the villagers don't exist in spectator mode (to spectators of arena).

It's made on purpose like this because shops don't really exists they are created only for people in survival mode. (Packet based, not a real entity)

**Comment by EastRane** [comment]
> SBA generate completely reset games-inventory with the new games, I guess that also affect the npc somehow, where were those strings located

Yes, it completely overwrites the file in the games-inventory folder when I use /sba generate. And there are lines of names/lores of inventory items and items with the join to the arena. If it's worth the coding time, imo it would make more sense to update only the arena list (or maybe have a separate command specifically for updating arenas within games-inventory, it makes even more sense), and leave all other lines as they are (to prevent user's custom non-join-arena-items inventory settings from reseting)

**Comment by AVLgames** [comment]
Only the gold and the Iron don't want to stack and the shopkeepers always are saying don't have enough iron/gold even when you have enough
but by Daimond and emerald it's working perfect.
